mk/turnkey/nginx-nodejs.mk

```diff
@@ -0,0 +1,4 @@
+COMMON_OVERLAYS += nodejs-nginx
+
+include $(FAB_PATH)/common/mk/turnkey/nodejs.mk
+include $(FAB_PATH)/common/mk/turnkey/nginx.mk
```

Comment:
Includes should probably go above the overlay here. It won't cause issues currently, but this means any overlays included from these 2 files will be applies AFTER `nodejs-nginx` which in the future may cause unexpected results.

Reply:
Hmm, I'm not 100% sure on this.

Looking at the scripts, `overlay/nodejs-nginx` only includes (node specific) nginx config (so doesn't matter when it's applied). `nodejs.mk` is the one that only includes `conf/nodejs` (which just sets up node and the `node` user) and `nginx.mk` includes nginx conf and overlay (neither of which are impacted either way by node specific stuff).

So long as overlays don't overwrite one another (which we possibly should have some text/check for? Perhaps a warning when overwriting any files ever, but an error if overlayed files are overwritten by other overlays? Not sure how realistic that might be though...) 

Anyway, having said all that, I don't have any real objection on changing it, but I don't think it's really required.

Reply:
I agree, however it seems to me un-intuitive that files depending on both `nginx` and `nodejs` to be written to the filesystem prior to the specific overlays for `nginx` and `nodejs` respectively.

And it seems possible to me that naive changes to one of these 3 overlays breaks the assumption that they don't interact. Re-ordering ensures that the most-specific overlay wins, rather than the least specific, which (at least I) would not expect had I made those naive changes.

-----

TL;DR agree with your case, but arrive at a different conclusion.